### ISSUE TITLE
broken alignment for captions in toctrees in ver. 0.5.1

### ISSUE THREAD

**Issue by AngryMaciek** [readthedocs/sphinx_rtd_theme#1112]
### Problem

I have been using the older version `0.4.3` and had a nice toctrees like below:
<img width="320" alt="Screenshot 2021-04-05 at 00 13 22" src="https://user-images.githubusercontent.com/19755308/113523008-75987400-95a5-11eb-8da9-c0669073a97b.png">

However, the search functionality was bugged so I updated to `0.5.1` and now I get:
<img width="328" alt="Screenshot 2021-04-05 at 00 13 33" src="https://user-images.githubusercontent.com/19755308/113523024-8ea12500-95a5-11eb-9f19-fcdca5fd3ed6.png">

And it seems to me that the captions for both toctree elements is not aligned anymore. Neither horizontally nor vertically (werid space between the caption and the first section, no space between last section of the previous toctree and a new one).

### Reproducible Project

```rst
.. toctree::
   :hidden:
   :maxdepth: 2
   :caption: General Information

   custom/installation
   custom/background
   custom/example-results
   custom/citing-contributing
   custom/changelog
   custom/license

.. toctree::
   :hidden:
   :maxdepth: 2
   :caption: API

   custom/users-api
   custom/internal-api
```

#### Expected Results

I would like to have a page layout just like for `0.4.3`.

### Environment Info

- Python Version: `3.7.4`
- Sphinx Version: `3.5.3`
- RTD Theme Version: `0.5.1`

**Comment by JWCook** [comment]
This seems to be related to the release of docutils 0.17: https://docutils.sourceforge.io/RELEASE-NOTES.html#id46

**Comment by Blendify** [comment]
This is caused by a recent release of docutils 0.17 if you pin 0.16 in a pip install requirements it should fix the issue.

rtd should fix this globally but that solution would work for you individually. I have a patch here that fixes this for the theme: #1113 if this gets merged/released it should fix the issue.

There also seems to be a fix in sphinx: https://github.com/sphinx-doc/sphinx/issues/9001 the fix for that might also fix this issue.

**Comment by Blendify** [comment]
See also https://github.com/readthedocs/sphinx_rtd_theme/issues/1111

**Comment by Blendify** [comment]
See the rtd documentation here on how to pin a dependency: https://docs.readthedocs.io/en/stable/guides/reproducible-builds.html#pinning-dependencies
The line in the requirements file should be:

```
docutils<0.17
```

I have reached out to the team at rtd, this should be fixed on the server-side / in the theme. If you do pin to an old version be prepared to revert that change once this has been fixed on a more global level.

**Comment by AngryMaciek** [comment]
Thank you very much for the prompt response!
Indeed, `docutils==0.17` solved the issue :)

**Comment by Blendify** [comment]
This has been fixed with RTD Theme Version `0.5.2`
